src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -140,6 +142,50 @@ static constexpr std::chrono::minutes MAX_LEDGER_AGE_ACQUIRE {1};
 // Don't acquire history if write load is too high
 static constexpr int MAX_WRITE_LOAD_ACQUIRE {8192};
 
+// Helper function for LedgerMaster::doAdvance()
+// Returns the minimum ledger sequence in SQL database, if any.
+boost::optional<LedgerIndex>
+minSqlSeq(Application& app)
+{
+    boost::optional<LedgerIndex> seq;
+    auto db = app.getLedgerDB().checkoutDb();
+    *db << "SELECT MIN(LedgerSeq) FROM Ledgers", soci::into(seq);
+    return seq;
+}
+
+// Helper function for LedgerMaster::doAdvance()
+// Return true if a ledger should be fetched from the network.
```

Comment:
Nice comment, but I have a small suggestion.  Consider replacing "a ledger" with "candidateLedger" in the comment.

Reply:
@scottschurr 
Spacing fixed, comment fixed.

For the unit tests for this specific behavior, there's no coverage for several generations of callers of shouldAcquire(), and none for what I can see for make sure that we maintain history. Quite difficult to implement.